src/private/MDMViewControllerTransitionContext.m

```diff
@@ -50,8 +50,17 @@ - (nonnull instancetype)initWithTransition:(nonnull id<MDMTransition>)transition
 #pragma mark - UIViewControllerAnimatedTransitioning
 
 - (NSTimeInterval)transitionDuration:(id<UIViewControllerContextTransitioning>)transitionContext {
-  // TODO(featherless): Expose a TransitionWithTiming protocol that allows the transition to
-  // customize this value.
+  if([_transition respondsToSelector:@selector(transitionDuration)]) {
+    NSTimeInterval duration;
+    NSMethodSignature *signature =
+    [[_transition class] instanceMethodSignatureForSelector:@selector(transitionDuration)];
+    NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:signature];
+    invocation.target = _transition;
+    invocation.selector = @selector(transitionDuration);
+    [invocation invoke];
+    [invocation getReturnValue:&duration];
+    return duration;
+  }
```

Comment:
Ah you don't have to do as much runtimery here: once you've verified that the `_transition` responds to the selector you can just cast _transition to your desired type. E.g.:

```objc
if ... {
  id<MDMTransitionWithCustomDuration> withCustomDuration = (id<MDMTransitionWithCustomDuration>)_transition;
  return [withCustomDuration transitionDuration...];
}
```